dataRetriever/interface.go

```diff
@@ -192,13 +192,27 @@ type ShardIdHashMap interface {
 }
 
 // Uint64SyncMapCacher defines a cacher-type struct that uses uint64 keys and sync-maps values
-type Uint64SyncMapCacher interface {
+//type Uint64SyncMapCacher_ interface {
+//	Clear()
+//	Get(nonce uint64) (ShardIdHashMap, bool)
+//	Merge(nonce uint64, src ShardIdHashMap)
+//	Remove(nonce uint64, shardId uint32)
+//	RegisterHandler(handler func(nonce uint64, shardId uint32, value []byte))
+//	Has(nonce uint64, shardId uint32) bool
+//	IsInterfaceNil() bool
+//}
+
+type HeadersPool interface {
 	Clear()
-	Get(nonce uint64) (ShardIdHashMap, bool)
-	Merge(nonce uint64, src ShardIdHashMap)
-	Remove(nonce uint64, shardId uint32)
-	RegisterHandler(handler func(nonce uint64, shardId uint32, value []byte))
-	Has(nonce uint64, shardId uint32) bool
+	Add(headerHash []byte, header data.HeaderHandler)
+	RemoveHeaderByHash(headerHash []byte)
```

Comment:
make the interface more simple. RemoveHeader would be enough, GetHeader as well. If it has hash as input, than it does by hash. If not, it does by nonce.